File: ios/Classes/style/AppInboxListViewStyle.swift
//
//  AppInboxListViewStyle.swift
//  exponea
//
//  Created by Adam Mihalik on 02/03/2023.
//

import Foundation
import UIKit

class AppInboxListViewStyle {
    var backgroundColor: String?
    var item: AppInboxListItemStyle?

    init(backgroundColor: String? = nil, item: AppInboxListItemStyle? = nil) {
        self.backgroundColor = backgroundColor
        self.item = item
    }

    func applyTo(_ target: UITableView) {
        if let backgroundColor = UIColor.parse(backgroundColor) {
            target.backgroundColor = backgroundColor
        }
        // note: 'item' style is used elsewhere due to performance reasons
    }
}

File: android/src/main/kotlin/com/exponea/style/AppInboxListViewStyle.kt
package com.exponea.style

import android.view.View
import com.exponea.asColor

data class AppInboxListViewStyle(
    var backgroundColor: String? = null,
    var item: AppInboxListItemStyle? = null
) {
    fun applyTo(target: View) {
        backgroundColor?.asColor()?.let {
            target.setBackgroundColor(it)
        }
        // note: 'item' style is used elsewhere due to performance reasons
    }
}
